lms/djangoapps/email_marketing/tasks.py

```diff
@@ -14,11 +15,12 @@
 from sailthru.sailthru_error import SailthruClientError
 
 log = logging.getLogger(__name__)
+SAILTHRU_LIST_CACHE_KEY = "email.marketing.cache"
 
 
 # pylint: disable=not-callable
 @task(bind=True, default_retry_delay=3600, max_retries=24)
-def update_user(self, sailthru_vars, email, new_user=False, activation=False):
+def update_user(self, sailthru_vars, email, site=None, new_user=False, activation=False):
```

Comment:
This line was what caused the errors during deployment today. Previously-queued tasks existed in rabbit, then newly deployed workers started to pick them up because (they think) they know how to handle `update_user` tasks. Unfortunately, the parameters the task was queued with do not match the parameters the worker expects, so an error is thrown.

The workaround to this are to either add a temporary shim method (which can be removed in the next deployment), or to change the method name anytime a parameter changes in a task signature. There are examples of both here, for the `recalculate_subsection_grades` task: https://github.com/edx/edx-platform/blob/master/lms/djangoapps/grades/tasks.py#L34-L55